### ISSUE TITLE
Certain StepFunctions exceptions uncatchable

### ISSUE THREAD

**Issue by aidan-** [spulec/moto#3005]
I am having an issue catching certain exceptions that are being thrown by only some of the StepFunction related methods.  I have managed to break it down to this simple example to illustrate my problem:

```
import pytest
import moto
from moto import mock_stepfunctions
import boto3

from botocore.exceptions import ClientError

def test_sfn_describe_state_machine():
    with mock_stepfunctions():
        with pytest.raises(ClientError):
            client = boto3.client("stepfunctions", region_name="ap-southeast-2")
            client.describe_state_machine(stateMachineArn="asd")

def test_sfn_start_execution():
    with mock_stepfunctions():
        with pytest.raises(ClientError):
            client = boto3.client("stepfunctions", region_name="ap-southeast-2")
            client.start_execution(stateMachineArn="asd")

```

Running this with pytest (or anything, that part isn't relevant) will result in the first test passing, with the `ClientError` successfully being caught.  The second test will fail.  From what I can see, the two methods are raising two different exceptions:

1st) `botocore.errorfactory.InvalidArn: An error occurred (InvalidArn) when calling the DescribeStateMachine operation: Invalid State Machine Arn: 'asd'`
2nd) `moto.stepfunctions.exceptions.InvalidArn: Invalid State Machine Arn: 'asd'`

From what I can see, they're both calling the same code path to validate the ARN, but only one is raising exception namespace to `moto`.  I've also had a look at the tests in the `moto` repository itself and added one that follows the pattern of my failing test, and it fails there too - so I don't think this is an environment issue.

I'm sure it's something obvious, but it's got my stumped at the moment.

**Comment by bblommers** [comment]
Hi @aidan-, our `describe_state_machine`-function catches errors properly, and returns the error details in the correct format:
https://github.com/spulec/moto/blob/master/moto/stepfunctions/responses.py#L81

The `start_execution`-function doesn't catch anything, and simply throws the Exception as is:
https://github.com/spulec/moto/blob/master/moto/stepfunctions/responses.py#L95

Thanks for raising it - marking it as a bug.

**Comment by bblommers** [comment]
This is available as of moto >= 1.3.15.dev871
